tests/functional/ingestion/IngestionReader.js

```diff
@@ -0,0 +1,247 @@
+const assert = require('assert');
+const async = require('async');
+const http = require('http');
+const MongoClient = require('mongodb').MongoClient;
+
+const dummyPensieveCredentials = require('./DummyPensieveCredentials.json');
+const dummySSHKey = require('./DummySSHKey.json');
+const zookeeper = require('../../../lib/clients/zookeeper');
+const IngestionReader = require('../../../lib/queuePopulator/IngestionReader');
+const { initManagement } = require('../../../lib/management/index');
+const dummyLogger = require('../../utils/DummyLogger');
+const testConfig = require('../../config.json');
+const { MetadataMock } = require('arsenal').testing.MetadataMock;
+const testPort = testConfig.extensions.ingestion.sources[0].port;
+const mockLogOffset = 2;
+
+/**
+ * The QueuePopulatorExtension class sends entries to kafka, but for testing
+ * purposes we may not have kafka setup for quick testing. This class will
+ * mock some of the functions that are used by the logReader classes, and will
+ * assert that the input from the logReaders is as expected.
+ */
+class TestIngestionQP {
+    constructor(params) {
+        this.config = params.config;
+        this.expectedEntry = params.expectedEntry;
```

Comment:
You should initialize this._batch in the constructor.

Reply:
https://github.com/scality/backbeat/commit/0c876539c83ef11190a41e140cab2a2756b2dc2e

I actually end up deleting `TestIngestionQP` since we're already checking through kafka, should I add that change to the pr here

Reply:
Ok you can just leave this as-is then and follow up with the changes in another PR.